core/protobuf/src/proto/dataserver/protocol.proto

```diff
@@ -75,6 +75,12 @@ message WriteRequest {
 
   optional bool eof = 6;
   optional bool cancel = 7;
+
+  // This is only applicable for ufs writes.
+  optional string ufs_path = 1000;
+  optional string owner = 1001;
+  optional string group = 1002;
+  optional int32 mode = 1003;
```

Comment:
Then maybe it is better to create another proto? something like this:

message WriteRequest {
    message CreateOptions (or UfsCreateOptions) {
     ...
    }
   optional CreateOptions ufs_create_options;
}

It is better to use small tags which is more efficient.